src/modules/adaptors/Balancer/BalancerPoolAdaptor.sol

```diff
@@ -0,0 +1,295 @@
+        if (actualAmountStaked >= amountIn) {
+            revert BalancerPoolAdaptor__MaxSlippageExceeded();
+        } // TODO: not sure about this error message or implementation
+
+        _revokeExternalApproval(address(bpt), _poolGauge);
+    }
+
+    /**
+     * @notice withdraw (unstake) BPT from respective pool gauge
+     * @param _poolGauge address of rewards gauge
+     * @param _amountOut number of bpts to withdraw
+     * @param _claim_rewards whether or not to claim pending rewards too (true == claim)
+     * @dev 1.) Interface custom as Balancer/Curve do not provide for liquidityGauges.
+     * @dev 2.) Strategists to provide _poolGauge (which they can get via off-chain querying methods)
+     */
+    function withdrawBPT(address _poolGauge, uint256 _amountOut, bool _claim_rewards) external {
+        ILiquidityGaugev3Custom poolGauge = ILiquidityGaugev3Custom(_poolGauge);
+
+        if (poolGauge.balanceOf(address(this)) >= _amountOut) {
+            revert BalancerPoolAdaptor__NotEnoughToWithdraw();
+        }
+
+        IERC20 bpt = IERC20(poolGauge.lp_token());
+        uint256 unstakedBPTBefore = bpt.balanceOf(address(this));
+
+        uint256 amountOut;
+        if (_amountOut == type(uint256).max) {
+            amountOut = stakedBefore; // deposited (staked) bpt at this point
+        } else {
+            amountOut = _amountOut;
+        }
+
+        bpt.approve(_poolGauge, amountOut);
+        poolGauge.withdraw(amountOut, _claim_rewards); // msg.sender should be cellar address bc delegateCall. TODO: see issue # <> and confirm address.
+
+        uint256 actualWithdrawn = bpt.balanceOf(address(this)) - unstakedBPTBefore;
+
+        if (actualWithdrawn <= _amountOut) {
+            revert BalancerPoolAdaptor__MaxSlippageExceeded();
+        } // TODO: not sure about this error message or implementation
+
+        _revokeExternalApproval(bpt, _poolGauge);
+    }
+
+    /**
+     * @notice claim rewards ($BAL and/or other tokens) from LP position
+     * @notice Have the strategist provide the address for the pool gauge
+     * @dev rewards are only accrue for staked positions
+     * @param _poolGauge address of rewards gauge
+     * @param _rewardToken address of reward token, if not $BAL, that strategist is claiming
+     * TODO: make all verbose text here into github issues
+     * TODO: add checks throughout function
+     */
+    function claimRewards(address _poolGauge, address _rewardToken) public {
+        ILiquidityGaugev3Custom poolGauge = ILiquidityGaugev3Custom(_poolGauge);
+
+        // TODO: checks - though, I'm not sure we need these. If cellar calls `claim_rewards()` and there's no rewards for them then... there are no explicit reverts in the codebase but I assume it reverts. Need to test it though: https://github.com/balancer/balancer-v2-monorepo/blob/master/pkg/liquidity-mining/contracts/gauges/ethereum/LiquidityGaugeV5.vy#L440-L450:~:text=if%20total_claimable%20%3E%200%3A
```

Comment:
Yeah I agree, I dont think we really need this check. The only times I can think when a strategist would be claiming "zero" rewards is
-They call `claimRewards` when they have nothing staked
-They call `claimRewards` twice in the same TX
-They deposit with the claim rewards bool set to true, then call `claimRewards` after

All these cases are user error so I think it is fine to handle this error on the user side(teach the strategist to not claim rewards multiple times during a call)

Reply:
Let's discuss this one further. I'll leave the comment for now during development.